### PR TITLE
fix: Monospacing errors in dashboards & charts

### PR DESCRIPTION
**Pull request by codemaster08240328**
### SUMMARY

On Charts and Dashboards, if the user has an error, the error message should be displayed in a monospace font so some of the formatting / details is not lost.

image

image

> After
image

image

### CODE REVIEW

**Comment by rusackas** [comment]
This has a side effect of monospacing ANY error in this dashboard/explore context. See screenshot below of one example.

image

@kasiazjc any objections to this? I think the pros outweigh the cons here.

**Comment by kasiazjc** [comment]
@rusackas does error always contain sql/code? If yes, I think that pros outweigh the cons if we cannot separate those. If no - this could create more confusion 🤔

File: superset-frontend/src/chart/Chart.jsx [hunk 1]
```diff
@@ -113,6 +113,14 @@ const RefreshOverlayWrapper = styled.div`
   justify-content: center;
 `;
 
+const MonospaceDiv = styled.div`
```
Comment (villebro) [inline superset-frontend/src/chart/Chart.jsx:116]: nit: I would perhaps consider naming this `MonospaceMessage`

**Review by villebro — APPROVED** [review]
Thanks for fixing this @codemaster08240328! This has been an annoyance for a long time, especially on BigQuery where error messages have been very difficult to decipher due to the type face being variable width.

@kasiazjc @rusackas due to the varied exception types we get from analytical databases it will often be difficult to distinguish between code-like exceptions that need monospacing and more general exceptions that don't. We can potentially add more fine-grained logic to the backend that tries to identify which messages need monospacing and which don't, but until we do I suggest merging this as-is, as I feel the problem this fixes is more severe than the side-effect it introduces.

**Pull request merged**